src/components/AddPet.jsx

```diff
@@ -51,10 +40,20 @@ export default function EditPet() {
   }
 
   return (
-    <PetForm
-      initialValues={initialValues}
-      onSubmit={onAddSubmit}
-      type="add"
-    ></PetForm>
+    <>
+      <Prompt
+        title="Are you sure?"
+        subtitle="If you leave the page, every change made will be lost!"
+        primaryButtonMessage="Yes! Leave the page!"
+        secondaryButtonMessage="No! Stay on page!"
+        when={formChanged}
```

Comment:
you can render the prompt inside the pet form ant then use the `pristine` or `dirty` property from `final-form` to know the form has changed, no need to keep that info on the state